src/commonMain/kotlin/au/id/micolous/metrodroid/transit/thess/ThessUltralightTransaction.kt

```diff
@@ -0,0 +1,70 @@
+/*
+ * ThessUltralightTransitFactory.kt
+ *
+ * Copyright 2024 apo-mak
+ *
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ */
+package au.id.micolous.metrodroid.transit.thess
+
+import au.id.micolous.metrodroid.multi.FormattedString
+import au.id.micolous.metrodroid.multi.Parcelize
+import au.id.micolous.metrodroid.time.MetroTimeZone
+import au.id.micolous.metrodroid.time.TimestampFull
+import au.id.micolous.metrodroid.transit.Transaction
+import au.id.micolous.metrodroid.transit.Trip
+
+@Parcelize
+data class ThessUltralightTransaction(
+        private val mTimestamp: TimestampFull?,
+        private val mIsEntry: Boolean,
+        private val mIsUsed: Boolean,
+        private val mIsSingleUse: Boolean,
+        private val mProductCode: Int) : Transaction {
+
+    override val timestamp get() = mTimestamp
+    
+    override val mode get() = Trip.Mode.METRO
+    
+    override val isTapOff get() = !mIsEntry
+    
+    override val isTapOn get() = mIsEntry
+    
+    override val fare get() = null // No fare stored in card
+    
+    override val machineID get() = null // Not implemented for now
+    
+    override fun getAgencyName(isShort: Boolean) = 
+        FormattedString("Thessaloniki Metro")
+    
+    override fun getRawFields(level: Int) = super.getRawFields(level).orEmpty() + listOfNotNull(
+        FormattedString("Product code") to FormattedString("0x${mProductCode.toString(16)}"),
+        if (mIsSingleUse)
```

Comment:
Can we have full ticketType here (0x80/0xf0). This is for raw fields, not for average end user

Reply:
I’ve implemented the change, but I’m not fully certain I understood the request correctly.
Could you please confirm that my implementation aligns with what you had in mind? :)